pjmedia/src/pjmedia/rtcp_xr.c

```diff
@@ -427,32 +427,49 @@ void pjmedia_rtcp_xr_rx_rtcp_xr( pjmedia_rtcp_xr_session *sess,
     if ((pkt_len + 1) > (size / 4))
 	return;
 
+    ck_len = pkt_len - 1;
+
     /* Parse report rpt_types */
     while ((pj_int32_t*)rb_hdr < (pj_int32_t*)pkt + pkt_len)
     {	
+	unsigned bl_len;
 	rb_len = pj_ntohs((pj_uint16_t)rb_hdr->length);
 
 	/* Just skip any block with length == 0 (no report content) */
 	if (rb_len) {
 	    switch (rb_hdr->bt) {
 		case BT_RR_TIME:
-		    rb_rr_time = (pjmedia_rtcp_xr_rb_rr_time*) rb_hdr;
+		    bl_len = sizeof(*rb_rr_time) / 4;
```

Comment:
I think the fix looks good. My concern here what happens if in the future a new type is added that's not divisible by 4? 
Will a similar check be sufficient to prevent an OOB? 
imagine a custom type with struct `pjmedia_rtcp_xr_rb_some_custom_type` that has sizeof() => 3 

a check that's copy and paste from this check wont work. This concern also applies if any of these struct changes in size and doesn't become divisible by 4 anymore.

I think getting the end of the buffer, and check that the `buffer + sizeof(*rb_rr_time)` is within the bounds of the buffer should be more general and work hopefully right regardless of the struct size

Does this sound reasonable?